lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -596,15 +636,47 @@ def test_user_no_auth(self, monkeypatch, db_session):
         assert e.value.http_status == HTTPStatus.UNAUTHORIZED
 
     @pytest.mark.parametrize(
-        "meta,error",
+        "meta,error,message",
         [
-            ("user.foo:1", HTTPStatus.UNAUTHORIZED),
-            ("x.y:1", HTTPStatus.BAD_REQUEST),
-            ("global.x=3", HTTPStatus.BAD_REQUEST),
-            ("dataset.notright:10", HTTPStatus.BAD_REQUEST),
+            (
+                "user.foo:1",
+                HTTPStatus.UNAUTHORIZED,
+                "Metadata key user.foo cannot be used by an unauthenticated client",
+            ),
+            ("x.y:1", HTTPStatus.BAD_REQUEST, "Metadata key 'x.y' is not supported"),
+            (
+                "global.x=3",
+                HTTPStatus.BAD_REQUEST,
+                "Missing terminator for 'global.x=3'",
+            ),
+            (
+                "dataset.notright:10",
+                HTTPStatus.BAD_REQUEST,
+                "Metadata key 'dataset.notright' is not supported",
+            ),
+            (
+                "'dataset.name:foo",
+                HTTPStatus.BAD_REQUEST,
+                'Bad quote termination in "\'dataset.name:foo"',
```

Comment:
What about a missing opening quote (e.g., `"dataset.name':foo"`), what result should that produce?

Reply:
Yeah, never search for a trailing quote without an opening quote, so this is trying to set the *key* named `dataset.name'`, which is an illegal key name.